src/main/java/io/odpf/firehose/sink/objectstorage/writer/remote/ObjectStorageChecker.java

```diff
@@ -13,40 +20,97 @@
 import java.util.concurrent.ExecutorService;
 import java.util.stream.Collectors;
 
+import static io.odpf.firehose.metrics.Metrics.*;
+import static io.odpf.firehose.sink.objectstorage.ObjectStorageMetrics.*;
+
 @AllArgsConstructor
 public class ObjectStorageChecker implements Runnable {
     private static final Logger LOGGER = LoggerFactory.getLogger(ObjectStorageChecker.class);
-    private final BlockingQueue<String> toBeFlushedToRemotePaths;
+
+    private final BlockingQueue<FileMeta> toBeFlushedToRemotePaths;
     private final BlockingQueue<String> flushedToRemotePaths;
     private final Set<ObjectStorageWriterWorkerFuture> remoteUploadFutures;
     private final ExecutorService remoteUploadScheduler;
     private final ObjectStorage objectStorage;
+    private final Clock clock;
+    private final Instrumentation instrumentation;
 
     @Override
     public void run() {
-        List<String> tobeFlushed = new ArrayList<>();
+        List<FileMeta> tobeFlushed = new ArrayList<>();
         toBeFlushedToRemotePaths.drainTo(tobeFlushed);
         remoteUploadFutures.addAll(tobeFlushed.stream()
-                .map(path -> new ObjectStorageWriterWorkerFuture(
-                        remoteUploadScheduler.submit(() -> objectStorage.store(path)), path)
+                .map(fileMeta -> new ObjectStorageWriterWorkerFuture(
+                        remoteUploadScheduler.submit(() -> {
+                            try {
+                                objectStorage.store(fileMeta.getFullPath());
+                            } catch (Exception e) {
```

Comment:
this looks very Ugly ! why not always throw ObjectStorageException ??

Reply:
if you want to do more than one line of code here, then make a separate class, rather than anonymous class here. 
put this code there and call scheulder.submit(new Worker())..